src/main/java/clueless/Game.java

```diff
@@ -67,25 +67,16 @@ public void changeTurn() {
 	 * @return nextPlayer
 	 */
 	public Player nextPlayer() {
-
-		// TODO: (MEGAN) complete logic
 		
-		/*
-		 * Normal Workflow:
-		 * 	1. find the Player in this.characterMap that does NOT have a 'wait' state
-		 * 			- you will want to use ' instanceof Player' logic to skip characters 
-		 * 			  that aren't Players
-		 * 			- you may assume only one player is NOT in 'wait' state when this method is called 
-		 * 			  during normal game runtime
-		 * 
-		 * 	2. determine what index this Player is at using the constant ordered array CHARACTER_TURN_ORDER (from ClueInterface)
-		 * 	3. grab the next player's name from CHARACTER_TURN_ORDER
-		 * 	4. use the grabbed player's name to return this.characterMap.get(<the player's name you grabbed>)
-		 * 
-		 * NOTE: you can see startingPlayer() for some guidance on looping over the ordered CHARACTER_TURN_ORDER array
-		 */
-
-		return null;
+		for (int i = 0; i < CHARACTER_TURN_ORDER.length; i++) 
+		{
+			if ((this.characterMap.get(CHARACTER_TURN_ORDER[i]) instanceof Player) && (Player.state != PLAYER_STATE_WAIT))					
```

Comment:
`(Player.state != PLAYER_STATE_WAIT)`

- this will not return you the state of the player you're trying to point to at index i.
- you should cast to Player after determining that the character in the map is instanceof Player
- since you're using && instead of &, it is probably OK to cast to Player directly in your if statement

Thus, you should do the following:

replace `Player.state` with `((Player) this.characterMap.get(CHARACTER_TURN_ORDER[i])).state`